### ISSUE TITLE
tbench build fails on RHEL 8,9,Ubuntu 22.04

### ISSUE THREAD

**Issue by bssrikanth** [intel/lkp-tests#191]
I am trying to get tbench run working with LKP.

Currently I am seeing failures in tbench build, which is currently fetched from repo `https://git.samba.org/sahlberg/dbench.git/?p=sahlberg/dbench.git;a=summary`, across RHEL8,RHEL 9 and Ubuntu [atleast 22.04].

```
In file included from nfsio.c:20:
nfs.h:9:10: fatal error: rpc/rpc.h: No such file or directory
    9 | #include <rpc/rpc.h>
      |          ^~~~~~~~~~~
compilation terminated.
make: *** [Makefile:43: nfsio.o] Error 1
==> ERROR: A failure occurred in build().
    Aborting...
Install tbench failed
```
Looks like rpc/rpc.h is no longer available in recent distros.

In order to get past this issue I modified pkg/tbench/PKGBUILD to fetch source from https://www.samba.org/ftp/tridge/dbench/ where I did not see the above mentioned build issue.

For tbench,dbench this approach works fine. But for dbench-cifs and dbench-nfs I see tests fail with below error, as pkg/dbench is softlinked to pkg/tbench.

```
2022-11-17 06:02:24 dbench -B nfs --server=127.0.0.1 --export=/export --protocol=tcp --loadfile=/usr/local/share/nfs_2.txt -t 60 128
Invalid option -B: unknown option
dbench version 4.00 - Copyright Andrew Tridgell 1999-2004
```

Looks like sahlberg/dbench version has backend support for cifs/nfs where as `tridge/dbench` does not. 

I think we can decouple dbench and tbench pkg/PKBUILD so that we can run atleast run `tbench` without hitting build issue seen with `sahlberg/dbench.git. `

Please let me know if you have any better solutions.

**Comment by rli9** [comment]
thanks @bssrikanth how about adding libc6-dev as depends, which include rpc/rpc.h?
```
$ apt-file search rpc/rpc.h
dietlibc-dev: /usr/include/diet/rpc/rpc.h
krb5-multidev: /usr/include/mit-krb5/gssrpc/rpc.h
libc6-dev: /usr/include/rpc/rpc.h
...
```

**Comment by bssrikanth** [comment]
thanks for pointers @rli9. On Ubuntu 22.04, I dont see rpc/rpc.h part of libc6-dev. Instead its been packaged with `libtirpc-dev:/usr/include/tirpc/rpc/rpc.h`. Same thing with fedora/centos/rhel I understand from https://fedoraproject.org/wiki/Changes/SunRPCRemoval.

Before using a different repository for tbench, which I mentioned in description, I had tried modifying existing pkg/tbench/PKGINSTALL to adopt to tirpc as below:

```
        CFLAGS=$(pkg-config --cflags smbclient)
        LDLIBS="${LDLIBS} -ltirpc"
        CFLAGS="${CFLAGS} -I/usr/include/tirpc"
        ./configure CFLAGS="${CFLAGS}" LDLIBS="${LDLIBS}" || return
```

But still tbench build had failed:

```
gcc -I/usr/include/samba-4.0 -I/usr/include/tirpc -Wall -W -I. -DVERSION=\"4.00\" -DDATADIR=\"/usr/local/share\" -D_FORTIFY_SOURCE=2  -c -o sockio.o sockio.c
Generating mount.h
rpcgen -h mount.x > mount.h
Generating nfs.h
rpcgen -h nfs.x > nfs.hmpiling nfsio.o
gcc -g -c nfsio.c -o nfsio.o
In file included from nfsio.c:20:
nfs.h:9:10: fatal error: rpc/rpc.h: No such file or directory
    9 | #include <rpc/rpc.h>
      |          ^~~~~~~~~~~
compilation terminated.
make: *** [Makefile:43: nfsio.o] Error 1
==> ERROR: A failure occurred in build().
    Aborting...
Install tbench failed
```

So I moved to using `https://www.samba.org/ftp/tridge/dbench/`. 

Please let me know if you find any ways to go around build issue with tirpc - tbench.

**Comment by bssrikanth** [comment]
Saw this https://github.com/intel/lkp-tests/commit/ef8a8e19b80ae369534a7ae5bc753b1a5bae5034 .. will test the same and get back. Thank you

**Comment by bssrikanth** [comment]
Have tested latest lkp source with tbench fix on RHEL 8.7, RHEL 9.0/1 and Ubuntu 22-04:

In summary: With updates to adaptation packages as mentioned below tbench runs fine on ubuntu 22-04 and  RHEL 9.x. But on RHEL 8.7 we still see tbench build issues, something which could be related to rpcgen version which comes with RHEL 8.x.


Changes with which tbench run fine on RHEL 9.x and Ubuntu 22.04:

```
diff --git a/distro/adaptation/ubuntu-22.04 b/distro/adaptation/ubuntu-22.04
index c3a1a6717..8a7cd2b32 100644
--- a/distro/adaptation/ubuntu-22.04
+++ b/distro/adaptation/ubuntu-22.04
@@ -7,3 +7,4 @@ libx32gcc1: libx32gcc-s1
 lib32gcc-dev: lib32gcc-12-dev
 libx32gcc-dev: libx32gcc-12-dev
 libipsec-mb0: libipsec-mb1
+libtirpc: libtirpc-common


+++ b/distro/adaptation/centos
@@ -131,7 +131,7 @@ libssl: openssl-devel
 libssl-dev: openssl-devel
 libstdc++6: libstdc++.x86_64
 libstdc++-7-dev: libstdc++-devel.x86_64 libstdc++-static.x86_64
-libtirpc: libtirpc1
+libtirpc1: libtirpc
 libtirpc-dev: libtirpc-devel
 libtool-bin: libtool
 libtsan0: libtsan.x86_64
@@ -229,3 +229,5 @@ xsltproc: libxslt
 nasm: nasm.x86_64
 libpython2.7: python2-libs
 libc-dev: glibc-devel
+libsmbclient-dev: libsmbclient-devel
+samba-dev: samba-devel
```

On RHEL 8.7:

Tbench fails to build with below error at first after including above mentioned changes in adaptation/centos:
```
Generating mount.h
rpcgen -h mount.x > mount.h
/bin/sh: line 1: rpcgen: command not found
make: *** [Makefile:51: mount.h] Error 127
==> ERROR: A failure occurred in build().
    Aborting...
Install tbench failed
```

After including rpcgen as prereq on RHEL 8.7 facing below tbench build issue:

```
libnfs.c: In function ‘nfsio_connect’:
libnfs.c:389:9: error: unknown type name ‘dirpath’
  389 |         dirpath mountdir=discard_const(export);
      |         ^~~~~~~
libnfs.c:28:28: warning: initialization of ‘int’ from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
   28 | #define discard_const(ptr) ((void *)((intptr_t)(ptr)))
      |                            ^
libnfs.c:389:26: note: in expansion of macro ‘discard_const’
  389 |         dirpath mountdir=discard_const(export);
      |                          ^~~~~~~~~~~~~
libnfs.c:391:9: error: unknown type name ‘mountres3’
  391 |         mountres3 *mountres;
      |         ^~~~~~~~~
libnfs.c:392:9: error: unknown type name ‘fhandle3’; did you mean ‘sa_handler’?
  392 |         fhandle3 *fh;
      |         ^~~~~~~~
      |         sa_handler
libnfs.c:416:13: warning: implicit declaration of function ‘inet_aton’ [-Wimplicit-function-declaration]
  416 |         if (inet_aton(server, &sin.sin_addr) == 0) {
      |             ^~~~~~~~~
libnfs.c:424:52: error: ‘MOUNT_PROGRAM’ undeclared (first use in this function); did you mean ‘NFS_PROGRAM’?
  424 |                 nfsio->clnt = clnttcp_create(&sin, MOUNT_PROGRAM, MOUNT_V3, &nfsio->s, 17*1024*1024, 17*1024*1024);
      |                                                    ^~~~~~~~~~~~~
      |                                                    NFS_PROGRAM
libnfs.c:424:52: note: each undeclared identifier is reported only once for each function it appears in
libnfs.c:424:67: error: ‘MOUNT_V3’ undeclared (first use in this function)
  424 |                 nfsio->clnt = clnttcp_create(&sin, MOUNT_PROGRAM, MOUNT_V3, &nfsio->s, 17*1024*1024, 17*1024*1024);
      |                                                                   ^~~~~~~~
libnfs.c:441:18: warning: implicit declaration of function ‘mountproc3_mnt_3’ [-Wimplicit-function-declaration]
  441 |         mountres=mountproc3_mnt_3(&mountdir, nfsio->clnt);
      |                  ^~~~~~~~~~~~~~~~
libnfs.c:441:17: warning: assignment to ‘int *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  441 |         mountres=mountproc3_mnt_3(&mountdir, nfsio->clnt);
      |                 ^
libnfs.c:447:21: error: request for member ‘fhs_status’ in something not a structure or union
  447 |         if (mountres->fhs_status != MNT3_OK) {
      |                     ^~
libnfs.c:447:37: error: ‘MNT3_OK’ undeclared (first use in this function)
  447 |         if (mountres->fhs_status != MNT3_OK) {
      |                                     ^~~~~~~
libnfs.c:448:87: error: request for member ‘fhs_status’ in something not a structure or union
  448 |                 printf("ERROR: Server returned error %d when trying to MNT\n",mountres->fhs_status);
      |                                                                                       ^~
libnfs.c:453:23: error: request for member ‘mountres3_u’ in something not a structure or union
  453 |         fh = &mountres->mountres3_u.mountinfo.fhandle;
      |                       ^~
libnfs.c:455:33: error: request for member ‘fhandle3_val’ in something not a structure or union
  455 |                               fh->fhandle3_val,
      |                                 ^~
libnfs.c:456:33: error: request for member ‘fhandle3_len’ in something not a structure or union
  456 |                               fh->fhandle3_len,
      |                                 ^~
libnfs.c: In function ‘nfsio_readdirplus’:
libnfs.c:1437:17: warning: implicit declaration of function ‘asprintf’; did you mean ‘vsprintf’? [-Wimplicit-function-declaration]
 1437 |                 asprintf(&new_name, "%s/%s", dir, e->name);
      |                 ^~~~~~~~
      |                 vsprintf
make: *** [Makefile:47: libnfs.o] Error 1
==> ERROR: A failure occurred in build().
    Aborting...
Install tbench failed
```

Version of rpcgen on RHEL 8.7 where we see tbench build issue: rpcgen-1.3.1-4.el8.x86_64
Version of rpcgen on RHEL 9.x where we do not see tbench build issue: rpcgen-1.4-9.el9.x86_64

**Comment by bssrikanth** [comment]
@rli9 need your suggestion on way ahead here.
